File: solutions/clojure/difference-of-squares/1/src/difference_of_squares.clj
(ns difference-of-squares)

(defn sum-of-squares [n]
  (cond (= n 1) 1
        :else (+ (sum-of-squares (- n 1)) (* n n)))
)

(defn sum-to [n]
  (cond (= n 1) 1
        :else (+ (sum-to (- n 1)) n))  
)

(defn square-of-sum [n]
  (* (sum-to n) (sum-to n))
)

(defn difference [n]
  (- (square-of-sum n) (sum-of-squares n))
)

File: solutions/cpp/difference-of-squares/1/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares {

int square_of_sum(const int n)
{
    int sum{};
    for (int i = 1; i <= n; i++)
        sum += i;
    return sum * sum;
}

int sum_of_squares(const int n)
{
    int sum{};
    for (int i = 1; i <= n; i++)
        sum += i * i;
    return sum;
}

int difference(const int n)
{
    return square_of_sum(n) - sum_of_squares(n);
}

}  // namespace difference_of_squares
